### PR TITLE
feat: add support for `Bytes` and `RawSlice` as inputs for Scripts

### LINKED ISSUE #1274: Support `Bytes` and `RawSlice` input for Scripts
### Motivation

Requirements:
- Vectors can be returned from Contracts https://github.com/FuelLabs/fuels-ts/issues/933
- Vectors can be sent to Predicates https://github.com/FuelLabs/fuels-ts/issues/1083
- Bytes can be sent and returned from Contracts https://github.com/FuelLabs/fuels-ts/issues/1188
- Bytes and RawSlice can be sent to Predicates https://github.com/FuelLabs/fuels-ts/issues/1205
- Related https://github.com/FuelLabs/fuels-ts/issues/1187

### Usage example

_No response_

### Possible implementations

- Add input for Bytes for scripts
- Add input for RawSlice for scripts

### PR DESCRIPTION
**Pull request by camsjams**
Closes #1274 

This _also_ ended up being a verification step, as my previous work in #1221 already allowed this to work :rocket: (see also #1273)

### CODE REVIEW

File: packages/fuel-gauge/src/bytes.test.ts [hunk 1]
```diff
@@ -104,4 +113,19 @@ describe('Bytes Tests', () => {
...
+    const scriptInstance = getScript<MainArgs, void>('script-bytes', wallet);
+
+    const bytes = [40, 41, 42];
+    const INPUT: Wrapper = {
+      inner: [bytes, bytes],
+      inner_enum: { Second: bytes },
+    };
+    await scriptInstance.functions.main(1, INPUT).call();
+
+    expect(true).toBe(true);
```
Comment (arboleya) [inline packages/fuel-gauge/src/bytes.test.ts:129]: Missing proper assertion.

File: packages/fuel-gauge/src/raw-slice.test.ts [hunk 2]
```diff
@@ -100,4 +119,19 @@ describe('Raw Slice Tests', () => {
...
+    const scriptInstance = getScript<MainArgs, void>('script-raw-slice', wallet);
+
+    const bytes = [40, 41, 42];
+    const INPUT: Wrapper = {
+      inner: [bytes, bytes],
+      inner_enum: { Second: bytes },
+    };
+    await scriptInstance.functions.main(1, INPUT).call();
+
+    expect(true).toBe(true);
```
Comment (arboleya) [inline packages/fuel-gauge/src/raw-slice.test.ts:135]: Missing proper assertion.

**Review by arboleya — COMMENTED** [review]
Also, several failing checks [here](https://github.com/FuelLabs/fuels-ts/actions/runs/6303098891).

checks

**Review by nedsalk — APPROVED** [review]

**Review by arboleya — APPROVED** [review]
Thank you for the adjustments.

It'd be great to have two extra tiny tests accounting for these branches:

https://github.com/FuelLabs/fuels-ts/blob/d8a3d87912768def242441c5732453f04cc4792b/packages/abi-coder/src/coders/byte.ts#L19-L21

https://github.com/FuelLabs/fuels-ts/blob/d8a3d87912768def242441c5732453f04cc4792b/packages/abi-coder/src/coders/raw-slice.ts#L18-L20

**Review by Torres-ssf — APPROVED** [review]

**Review by arboleya — APPROVED** [review]

**Pull request merged**